sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/PartitionKeyBuilder.java

```diff
@@ -0,0 +1,90 @@
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.implementation.Undefined;
+import com.azure.cosmos.implementation.routing.PartitionKeyInternal;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PartitionKeyBuilder {
+    private List<Object> partitionKeyValues;
+
+    /**
+     * Constructor. CREATE a new instance of the PartitionKeyBuilder object.
+     */
+    public PartitionKeyBuilder()
+    {
+        this.partitionKeyValues = new ArrayList<Object>();
+    }
+
+    public PartitionKeyBuilder Add(String value)
+    {
+        this.partitionKeyValues.add(value);
+        return this;
+    }
+
+    public PartitionKeyBuilder Add(double value)
+    {
+        this.partitionKeyValues.add(value);
+        return this;
+    }
+
+    public PartitionKeyBuilder Add(boolean value)
+    {
+        this.partitionKeyValues.add(value);
+        return this;
+    }
+
+    public PartitionKeyBuilder AddNullValue()
+    {
+        this.partitionKeyValues.add(null);
+        return this;
+    }
+
+    public PartitionKeyBuilder AddNoneValue()
```

Comment:
Code doc please. 

For None please cover it explicitly on targeted scenarios.

Reply:
Null and None dis-ambiguration is very important in docs.

Reply:
Added code comments..